### PR TITLE
[SYCL] Add empty node implementation

### PR DESCRIPTION
**Pull request by reble**

### CODE REVIEW

File: sycl/include/sycl/ext/oneapi/experimental/graph.hpp [hunk 1]
```diff
@@ -55,7 +55,7 @@ class __SYCL_EXPORT command_graph {
   command_graph(const property_list &propList = {});
 
   // Adding empty node with [0..n] predecessors:
-  node add(const std::vector<node> &dep = {});
+  node add(const std::vector<node> &dep = {}) { return add_impl(dep); }
```
Comment (EwanC) [inline sycl/include/sycl/ext/oneapi/experimental/graph.hpp:58]: I don't think we have a method for `add_impl` which matches this usage, when I try to use the empty node I see this error.
```cpp
/home/ewan/Development/dpc++/build_release/bin/../include/sycl/ext/oneapi/experimental/graph.hpp:58:68: error: too few arguments to function call, expected 2, have 1
  node add(const std::vector<node> &dep = {}) { return add_impl(dep); }
                                                       ~~~~~~~~    ^

/home/ewan/Development/dpc++/build_release/bin/../include/sycl/ext/oneapi/experimental/graph.hpp:110:8: note: 'add_impl' declared here
  node add_impl(std::function<void(handler &)> cgf,
       ^
```

Can we add a test using this empty node API? Would help to catch this.

Comment (reble) [reply sycl/include/sycl/ext/oneapi/experimental/graph.hpp:58]: Agreed makes sense. I think I had a test in the initial PR #62 that just got lost by cherry picking the changes (including the implementation). Will include that.

**Review by EwanC — APPROVED** [review]

File: sycl/test/graph/graph-explicit-empty.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+#include <iostream>
```
Comment (Bensuo) [inline sycl/test/graph/graph-explicit-empty.cpp]:
```suggestion
```
`iostream` is not required anymore.

File: sycl/test/graph/graph-explicit-empty.cpp [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    h.parallel_for(sycl::range<1>{n}, [=](sycl::id<1> idx) {
+      size_t i = idx;
+      arr[i] = 1;
+    });
+  }, {empty});
+
+  auto executable_graph = g.finalize(q.get_context());
+
+  q.submit([&](sycl::handler &h) { h.ext_oneapi_graph(executable_graph); }).wait();
+
```
Comment (Bensuo) [inline sycl/test/graph/graph-explicit-empty.cpp:41]: We should have an `assert` verifying the results of this example in keeping with the others.

**Review by Bensuo — APPROVED** [review]
LGTM just a few minor comments. I think there will be changes or additions to this needed for #100 to hook up the edges correctly, but where exactly those changes will be depends on the order of merging I suppose.

File: sycl/test/graph/graph-explicit-empty.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  sycl::property_list properties{
+      sycl::property::queue::in_order{},
+      sycl::ext::oneapi::property::queue::lazy_execution{}};
+
+  sycl::queue q{sycl::gpu_selector_v, properties};
+
+  sycl::ext::oneapi::experimental::command_graph g;
+
+  const size_t n = 10;
+  float *arr = sycl::malloc_device<float>(n, q);
```
Comment (reble) [inline sycl/test/graph/graph-explicit-empty.cpp]:
```suggestion
  const size_t n = 10;
  float *h_arr = sycl::malloc_host<float>(n, q);
  float *arr = sycl::malloc_device<float>(n, q);
```

File: sycl/test/graph/graph-explicit-empty.cpp [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    h.parallel_for(sycl::range<1>{n}, [=](sycl::id<1> idx) {
+      size_t i = idx;
+      arr[i] = 1;
+    });
+  }, {empty});
+
+  auto executable_graph = g.finalize(q.get_context());
+
+  q.submit([&](sycl::handler &h) { h.ext_oneapi_graph(executable_graph); }).wait();
+
```
Comment (reble) [inline sycl/test/graph/graph-explicit-empty.cpp:41]:
```suggestion

  q.memcpy(&(h_arr[0]), arr, n).wait();

  for (int i = 0; i < n; i++)
    assert(h_arr[i] == 1.0f);

  sycl::free(h_arr, q);
```
